src/dimension.c

```diff
@@ -478,17 +488,26 @@ dimension_insert_relation(Relation rel, int32 hypertable_id,
 			DirectFunctionCall1(namein, CStringGetDatum(get_func_name(partitioning_func)));
 		values[Anum_dimension_partitioning_func_schema - 1] =
 			DirectFunctionCall1(namein, CStringGetDatum(get_namespace_name(pronamespace)));
+	}
+	else
+	{
+		nulls[Anum_dimension_partitioning_func - 1] = true;
+		nulls[Anum_dimension_partitioning_func_schema - 1] = true;
+	}
+
+	if (num_slices > 0)
+	{
+		/* Closed (hash) dimension */
 		values[Anum_dimension_num_slices - 1] = Int16GetDatum(num_slices);
 		values[Anum_dimension_aligned - 1] = BoolGetDatum(false);
 		nulls[Anum_dimension_interval_length - 1] = true;
 	}
 	else
 	{
+		/* Open (time) dimension */
```

Comment:
Similarly, should we assert `num_slices == 0` here?

Reply:
It can't be greater than zero (enabled), since otherwise we wouldn't really get here. But we can add the `Assert` if you think it improves readability.